airbyte-cdk/python/unit_tests/sources/message/test_repository.py

```diff
@@ -90,14 +89,6 @@ def test_given_unknown_log_level_as_threshold_when_log_message_then_allow_messag
         repo.log_message(Level.DEBUG, lambda: {"message": "this is a log message"})
         assert list(repo.consume_queue())
 
-    def test_given_unknown_log_level_for_log_when_log_message_then_raise_error(self):
```

Comment:
it is not still possible to pass an invalid level?

Reply:
originally we used pydantic model validation (internal) and no other level can be passed to the model, except the ones mentioned in Protocol, see https://github.com/airbytehq/airbyte/blob/32fdd7fd7222e10565568d7663bad8dd40278600/airbyte-cdk/python/airbyte_cdk/sources/message/repository.py#L80.

But, we still got a log warning message in case level is not supported:
https://github.com/airbytehq/airbyte/blob/32fdd7fd7222e10565568d7663bad8dd40278600/airbyte-cdk/python/airbyte_cdk/sources/message/repository.py#L35-L37

Should we raise an exception here instead?

Reply:
raising an exception or at least logging the warning seems like the right thing to me to avoid surprises